pkg/pkgmgr/apk.go

```diff
@@ -165,6 +165,12 @@ func (am *apkManager) upgradePackages(ctx context.Context, updates unversioned.U
 	// TODO: Add support for custom APK config
 	apkUpdated := am.config.ImageState.Run(llb.Shlex("apk update"), llb.WithProxy(utils.GetProxy()), llb.IgnoreCache).Root()
 
+	// If updating all packages, check for upgrades before proceeding with patch
+	if updates == nil {
+		checkUpgradable := `sh -c "apk list 2>/dev/null | grep -q "upgradable" || exit 1"`
```

Comment:
can we use `apk list --upgradeable`? this should be a smaller list

Reply:
when testing with some images like `alpine:3.7.3`, `apk list `showed updates, and so did running `apk upgrade`. But running `apk list --upgradable` did not. The flag wasn't working as expected so it seemed apk list would be more accurate.

Reply:
Ok interesting, might be a bug with older version of apk